When creating a file in the SD card, set its creation time via RTC

fiam opened the pull request:
If the RTC doesn't have a valid time, the previous default value
as defined by AFATFS_DEFAULT_FILE_DATE and AFATFS_DEFAULT_FILE_TIME
is used instead.

fiam committed:
When creating a file in the SD card, set its creation time via RTC

If the RTC doesn't have a valid time, the previous default value
as defined by AFATFS_DEFAULT_FILE_DATE and AFATFS_DEFAULT_FILE_TIME
is used instead.

digitalentity commented on src/main/io/asyncfatfs/asyncfatfs.c:
```diff
@@ -2605,8 +2607,12 @@ static void afatfs_createFileContinue(afatfsFile_t *file)
+                if (rtcGetDateTime(&now)) {
+                    entry->creationDate = FAT_MAKE_DATE(now.year, now.month, now.day);
+                    entry->creationTime = FAT_MAKE_TIME(now.hours, now.minutes, now.seconds);
+                }
```
Shouldn't we move 
```
                 entry->creationDate = AFATFS_DEFAULT_FILE_DATE;
                  entry->creationTime = AFATFS_DEFAULT_FILE_TIME;
```
to an `else` clause?

fiam replied:
Well, I incorrectly assumed that not using the `else` would save a bit of flash since the macro should resolve to a constant a compile time. However, using the else saves 18 bytes of TEXT, so I'll amend it.

fiam committed:
Use a else for the default FAT timestamp

Rather than setting the default values unconditionally. Saves
18 bytes of flash.
